jax/numpy/lax_numpy.py

```diff
@@ -1338,9 +1333,7 @@ def array_equal(a1, a2):
     a1, a2 = asarray(a1), asarray(a2)
   except Exception:
     return False
-  if a1.shape != a2.shape:
-    return False
-  return asarray(a1==a2).all()
+  return shape(a1) == shape(a2) and all(asarray(a1 == a2))
 
```

Comment:
Is calling all(...) correct for traced values?

Reply:
Yes it should be, because that’s jax.numpy.all rather than the builtin. That said I didn’t double check how thorough the tests are...

Reply:
Looks to me like the tests cover this!